File: C++/Fibonacci/fibonacci.cpp
#include <iostream>
using namespace std;

void fibonacci(int length) {
  int size = length;
  int* result = new int[size];

  for (int i = 2; i < length; ++i) {
    result[i] = result[i - 1] + result[i - 2];
  }

  cout << result << endl;
}

// Main() function: Where program execution begins
int main() {
  fibonacci(5);

  return 0;
}

File: Rust/Fibonacci/fibonacci.rs
fn main() {
  println!("{:?}", fibonacci(5));
}

fn fibonacci(length: i32) -> Vec<i32> {
  if length <= 0 {
    return vec![];
  }

  if length == 1 {
    return vec![1];
  }

  let mut result: Vec<i32> = vec![1, 1];

  for n in 2..length as usize {
    result.push(result[n - 1] + result[n - 2]);
  }

  return result;
}